dsgrid/registry/project_registry_manager.py

```diff
@@ -641,16 +686,39 @@ def _run_checks(self, config: ProjectConfig):
         dims = [x for x in config.iter_dimensions()]
         check_uniqueness((x.model.name for x in dims), "dimension name")
         check_uniqueness((x.model.display_name for x in dims), "dimension display name")
-        check_uniqueness(
-            (getattr(x.model, "cls") for x in config.model.dimensions.base_dimensions),
-            "dimension cls",
-        )
+        self._check_base_dimensions(config)
+        self._check_base_dimension_record_uniqueness(config)
+
         for dataset_id in config.list_unregistered_dataset_ids():
             for field in RequiredDimensionRecordsModel.model_fields:
                 # This will check that all dimension record IDs listed in the requirements
                 # exist in the project.
                 config.get_required_dimension_record_ids(dataset_id, DimensionType(field))
 
+    def _check_base_dimensions(self, config: ProjectConfig) -> None:
+        found_time = False
+        for dim in config.list_base_dimensions():
+            if dim.model.dimension_type == DimensionType.TIME:
+                if found_time:
+                    msg = "Only one time dimension is allowed in a project."
+                    raise DSGInvalidDimension(msg)
+                found_time = True
+
+    def _check_base_dimension_record_uniqueness(self, config: ProjectConfig) -> None:
+        for dimension_type in DimensionType:
+            if dimension_type == DimensionType.TIME:
+                continue
+            record_ids: set[str] = set()
+            for dim in config.list_base_dimensions_with_records(dimension_type):
+                ids = get_unique_values(dim.get_records_dataframe(), "id")
+                if intersect := record_ids.intersection(ids):
+                    msg = (
+                        "Record IDs across base dimensions of the same type must be unique: "
+                        f"{sorted(intersect)}"
+                    )
```

Comment:
I don't love this requirement. For example, with the buildings data it seems reasonable for the energy data to be `electricity_heating`, `electricity_cooling` with units of kWh/MWh and for the shed data to be `electricity_heating`, `electricity_cooling` with units of kW/MW. 

However, I also see the point that people are less likely to mess up if the latter are instead `electricity_heating_shed` and `electricity_cooling_shed`.

Reply:
If we keep this requirement, does it also provide a way to figure out what the base dimension actually is when multiple mappings are returned for a supplemental dimension?

Reply:
No longer required.